packages/node_modules/@node-red/nodes/core/function/90-exec.js

```diff
@@ -137,6 +145,16 @@ module.exports = function(RED) {
                             node.error(code,RED.util.cloneMessage(msg));
                         }
                     });
+
+                    if (node.addpay && node.addpayTo == "stdin")  {
+                        var value = RED.util.getMessageProperty(msg, node.addpay);
+                        if (value !== undefined) {
+                            var stdinStream = new stream.Readable();
+                            stdinStream.push(value);  // Add data to the internal queue for users of the stream to consume
```

Comment:
```suggestion
                            if (!Buffer.isBuffer(value) && (typeof value) !== 'string') {
                                // Ensure value is a string or buffer - JSON.stringify anything else
                                value = JSON.stringify(value)
                            }
                            stdinStream.push(value);  // Add data to the internal queue for users of the stream to consume
```

Reading up on the docs for `stream.Readable.push` I think we should be a bit more guarded as to what types are passed in here. They are expected to be Buffers or Strings (mostly). Above suggestion is an untested way to ensure the type is good.